Description: Use distribute by default.
Author: Barry Warsaw <barry@python.org>
Forwarded: no

Index: b/virtualenv.py
===================================================================
--- a/virtualenv.py
+++ b/virtualenv.py
@@ -883,7 +883,7 @@
         'VIRTUALENV_SETUPTOOLS to make it the default ')
 
     # Set this to True to use distribute by default, even in Python 2.
-    parser.set_defaults(use_distribute=False)
+    parser.set_defaults(use_distribute=True)
 
     default_search_dirs = file_search_dirs()
     parser.add_option(
